### DISCUSSION TITLE
nvidia env variables to try for VSCode/electron apps

### DISCUSSION THREAD

**Discussion by CVLover** [prasanthrangan/hyprdots#654] [Q&A]
Heres some env and misc edits i had to do to fix that
gpu is a 4070Ti

env = GBM_BACKEND,nvidia-drm
env = LIBVA_DRIVER_NAME,nvidia
env = __GLX_VENDOR_LIBRARY_NAME,nvidia
env = __GL_VRR_ALLOWED,1
env = WLR_NO_HARDWARE_CURSORS,1
env = WLR_DRM_NO_ATOMIC,1
env = CLUTTER_BACKEND=wayland
env = WLR_EGL_NO_MODIFIERS,0
env = WLR_RENDERER_ALLOW_SOFTWARE,1
env = MOZ_ENABLE_WAYLAND,1
env = MOZ_DISABLE_RDD_SANDBOX,1
env = MOZ_DBUS_REMOTE,1
env = __GL_MaxFramesAllowed,1
env = XWAYLAND_NO_GLAMOR,1
env = __VK_LAYER_NV_optimus,NVIDIA_only
env = NVD_BACKEND,direct
env = WLR_DRM_DEVICES,/dev/dri/card1
env = __GL_GSYNC_ALLOWED,1
env = PROTON_ENABLE_NGX_UPDATER,1
env = WLR_USE_LIBINPUT,1
env = ELECTRON_ARGS="--enable-features=UseOzonePlatform --ozone-platform=wayland"

**Comment by prasanthrangan** [comment]
Not sure if its a good option to enable these by default, as `env = GBM_BACKEND,nvidia-drm` tend to cause boot (not grub, hyprland) issues
also `WLR_DRM_DEVICES,/dev/dri/card1` this would change from system to system...
if you have issues with vscode, try using `--disable-gpu` option, it does fix all the lags with vscode in my system

**Comment by henriquejustee** [comment]
@prasanthrangan, the 'env = XWAYLAND_NO_GLAMOR,1' helps with some Electron-based software like Insomnia, VSCode, and Discord. However, it prevents me from starting other software like Steam. I'm experiencing lagging, flickering apps, and gaming issues without this problematic env option, most likely due to a hardware acceleration problem. Do you know of any fixes? I'm using Nvidia, unfortunately-

**Comment by prasanthrangan** [comment]
Hi @HenriqueJuste01 i have same issues as well with nvidia hardware acceleration for Electron apps. for me `--disable-gpu` launch option just works fine.

I'll move this to discussion as it will helpful to other users.

**Reply by t101804** [reply]
any alternatives? i need to run electron app with gpu enabled

**Reply by t101804** [reply]
any issue to fix it? example in discord?
